h2o-automl/src/test/java/ai/h2o/automl/leaderboard/LeaderboardTest.java

```diff
@@ -0,0 +1,167 @@
+    } finally {
+      if (lb != null){
+        lb.remove();
+      }
+      eventLog.remove();
+      if (model != null) {
+        model.deleteCrossValidationModels();
+        model.delete();
+      }
+      if( fr != null) {
+        fr.delete();
+      }
+    }
+  }
+
+  @Test
+  public void test_leaderboard_table_with_extensions() {
+    List<Keyed> removables = new ArrayList<>();
+    try {
+      String target = "CAPSULE";
+      final Frame fr = parse_test_file("./smalldata/logreg/prostate_train.csv").toCategoricalCol(target);  removables.add(fr);
+      GBMModel.GBMParameters parms = new GBMModel.GBMParameters();
+      parms._train = fr._key;
+      parms._nfolds = 3;
+      parms._seed = 1;
+      parms._response_column = "CAPSULE";
+      GBM job = new GBM(parms);
+      Model model = job.trainModel().get(); removables.add(model);
+
+      EventLog eventLog = EventLog.getOrMake(dummy); removables.add(eventLog);
+      Leaderboard lb = Leaderboard.getOrMake("leaderboard_with_ext", eventLog, null, null); removables.add(lb);
+      lb.setExtensionsProvider(new LeaderboardExtensionsProvider() {
+        @Override
+        public LeaderboardCell[] createExtensions(Model model) {
+          return new LeaderboardCell[] {
+                  new TrainingTime(model),
+                  new ScoringTimePerRow(model, null, fr),
+          };
+        }
+      });
+      lb.addModel(model._key);
+      TwoDimTable lb_table = lb.toTwoDimTable();
+      assertEquals(1, lb_table.getRowDim()); // one model, one row
+      assertEquals(7, lb_table.getColDim()); // model_id + 6 binomial metrics
+      assertEquals("Leaderboard for project leaderboard_with_ext", lb_table.getTableHeader());
+      assertEquals("models sorted in order of auc, best first", lb_table.getTableDescription());
+      assertArrayEquals(new String[] {"model_id", "auc", "logloss", "aucpr", "mean_per_class_error", "rmse", "mse"},
+              lb_table.getColHeaders());
+      assertArrayEquals(new String[] {"string", "double", "double", "double", "double", "double", "double"},
+              lb_table.getColTypes());
+      assertArrayEquals(new String[] {"%s", "%.6f", "%.6f", "%.6f", "%.6f", "%.6f", "%.6f"},
+              lb_table.getColFormats());
+
+      TwoDimTable lb_table_ext = lb.toTwoDimTable(LeaderboardExtensionsProvider.ALL);
+      assertEquals(7+2, lb_table_ext.getColDim());
+      assertArrayEquals(new String[] {"model_id", "auc", "logloss", "aucpr", "mean_per_class_error", "rmse", "mse", "training_time_ms", "predict_time_per_row_ms"},
+              lb_table_ext.getColHeaders());
+      assertArrayEquals(new String[] {"string", "double", "double", "double", "double", "double", "double", "long", "double"},
+              lb_table_ext.getColTypes());
+      assertArrayEquals(new String[] {"%s", "%.6f", "%.6f", "%.6f", "%.6f", "%.6f", "%.6f", "%s", "%.6f"},
+              lb_table_ext.getColFormats());
+      assertTrue(lb_table_ext.get(0, 7) instanceof Long);
```

Comment:
@seb-h2o can I ask you to use self descriptive variables for these numbers ( 7, 8 etc) ? would make it easier to read test. Or comments, but imho variables is a better way to avoid repetitions of comments

Reply:
will do